### PR TITLE
Set work coordinate

### PR DESCRIPTION
**Pull request by breiler**
Added functionality for setting work coordinates requested in #977.

It's also possible to use simple mathematical expressions such as the following:
set_work_coords

### CODE REVIEW

File: ugs-core/src/com/willwinder/universalgcodesender/model/GUIBackend.java [hunk 1]
```diff
@@ -808,7 +812,33 @@ public void sendMessageForConsole(String msg) {
...
     }
-    
+
+    @Override
+    public void setWorkPosition(Axis axis, double position) throws Exception {
+        controller.setWorkPosition(axis, position);
+    }
+
+    @Override
+    public void setWorkPositionUsingExpression(final Axis axis, final String expression) throws Exception {
```
Comment (winder) [inline ugs-core/src/com/willwinder/universalgcodesender/model/GUIBackend.java]: Wow, this is really cool. I had no idea you could do something like this.

**Comment by jahnj0584** [comment]
Wicked. How complex of equations can you do? I assume it only modifies the
current WCS?

**Comment by winder** [comment]
If the expression starts with +, -, * or / the current value is inserted to the left, otherwise it evaluates whatever you put in directly.

So if the axis is at 10, "+ 2" is evaluated as "10 + 2". But if you type in "2 + 2" it evaluates to 4 even though you are at 10.

The scripting language is javascript so you can put whatever you want in there. I tried the `<<` bitshift operator and it worked, you can chain them together so if you are at 11, `* 1 << 1` it evaluates as `11*1<<1` which turns into 22.

**Comment by jahnj0584** [comment]
But you could do something like "/2 -29" when you want it to be 29mm away
from the center?

**Comment by flounderscore** [comment]
How can I set the current position to a negative value? It seems like it
would attempt do perform math instead.

**Comment by winder** [comment]
@jahnj0584 it would depend on your order of operations, but I think so. Let us know, I just merged it in :)

**Comment by winder** [comment]
@flounderscore you're right, you would have to do `10 * -1`

**Comment by winder** [comment]
@flounderscore or zero out the axis first so that the math is right.

**Comment by flounderscore** [comment]
Any chance to make the math feature optional? All my coordinates are typically <0.

File: ugs-core/src/com/willwinder/universalgcodesender/model/GUIBackend.java (hunk 1 above)
Comment (breiler) [reply ugs-core/src/com/willwinder/universalgcodesender/model/GUIBackend.java]: Me neither, google was my friend! 😄

File: ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java [hunk 2]
```diff
@@ -187,7 +187,12 @@ public void resetCoordinatesToZero() throws Exception {
     public void resetCoordinateToZero(final Axis coord) throws Exception {
```
[the remark this answers is not in the document]
Comment (breiler) [reply ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java]: That is a good idea!

**Pull request merged**